### PR TITLE
pull: handle unneeded and disabled bottles.

### PR DESCRIPTION
**Pull request by mikemcquaid**
CC @xu-cheng @bfontaine @DomT4 for review. I've used it a bit and it works well. Would be good to merge ASAP so I can use it.

### CODE REVIEW

File: Library/Homebrew/cmd/pull.rb [hunk 1]
```diff
@@ -128,8 +128,12 @@ def pull
         end
       end
 
-      unless ARGV.include? "--bottle"
-        changed_formulae.each do |f|
+      fetch_bottle = false
+      changed_formulae.each do |f|
+        if ARGV.include? "--bottle"
+          fetch_bottle ||= !!f.bottle
```
Comment (dunn) [inline Library/Homebrew/cmd/pull.rb]: Since it's explicitly set to false, won't these conditional assignments always be ignored?

Comment (mikemcquaid) [inline Library/Homebrew/cmd/pull.rb]: Nope; `||=` sets it if the value is currently falsey.

Comment (dunn) [inline Library/Homebrew/cmd/pull.rb]: TIL!

Comment (mikemcquaid) [inline Library/Homebrew/cmd/pull.rb]: It's equivalent of `fetch_bottle = fetch_bottle || !!f.bottle`

**Comment by DomT4** [comment]
I'd be tempted to shout at us that the bottle was disabled for x formula in the pull, but I'm not that fussed about it if you're against. Happy otherwise.

**Comment by mikemcquaid** [comment]
@DomT4 Updated. Let me know what you think.

**Comment by xu-cheng** [comment]
:+1:

**Comment by bfontaine** [comment]
I was sleeping at this time but :+1: :wink:

**Comment by DomT4** [comment]
Thanks @mikemcquaid, the changed the language is :100:. Appreciate the tweak!

**Pull request closed without merging**